components/eventing-controller/controllers/subscription/nats/unit_test.go

```diff
@@ -292,18 +359,61 @@ func createFakeClient(g *GomegaWithT) client.WithWatch {
 	return fake.NewClientBuilder().WithScheme(scheme.Scheme).Build()
 }
 
-func getSubscriptionWithConditionsAndStatus(conditions []eventingv1alpha1.Condition, status bool, sink string) *eventingv1alpha1.Subscription {
-	return &eventingv1alpha1.Subscription{
+func NewSubscription(options ...func(subscription *eventingv1alpha1.Subscription)) *eventingv1alpha1.Subscription {
+	sub := &eventingv1alpha1.Subscription{
 		ObjectMeta: v1.ObjectMeta{
 			Name:      "test",
 			Namespace: namespaceName,
 		},
-		Spec: eventingv1alpha1.SubscriptionSpec{
-			Sink: sink,
-		},
-		Status: eventingv1alpha1.SubscriptionStatus{
-			Conditions: conditions,
-			Ready:      status,
-		},
 	}
+	for _, o := range options {
+		o(sub)
+	}
+	return sub
+}
+func WithSink(sink string) func(subscription *eventingv1alpha1.Subscription) {
+	return func(sub *eventingv1alpha1.Subscription) {
+		sub.Spec.Sink = sink
+	}
+}
+func WithConditions(conditions []eventingv1alpha1.Condition) func(subscription *eventingv1alpha1.Subscription) {
+	return func(sub *eventingv1alpha1.Subscription) {
+		sub.Status.Conditions = conditions
+	}
+}
+func WithStatus(status bool) func(subscription *eventingv1alpha1.Subscription) {
+	return func(sub *eventingv1alpha1.Subscription) {
+		sub.Status.Ready = status
+	}
+}
+func WithFinalizers(finalizers []string) func(subscription *eventingv1alpha1.Subscription) {
+	return func(sub *eventingv1alpha1.Subscription) {
+		sub.ObjectMeta.Finalizers = finalizers
+	}
+}
+
+func fetchTestSubscription(ctx context.Context, r *Reconciler) (eventingv1alpha1.Subscription, error) {
+	var fetchedSub eventingv1alpha1.Subscription
+	err := r.Client.Get(ctx, types.NamespacedName{
+		Name:      "test",
+		Namespace: namespaceName,
+	}, &fetchedSub)
+	return fetchedSub, err
+}
+
+func ensureFinalizerMatch(g *GomegaWithT, subscription *eventingv1alpha1.Subscription, expectedFinalizers []string) {
+	if len(expectedFinalizers) == 0 {
+		g.Expect(subscription.ObjectMeta.Finalizers).Should(BeEmpty())
+	} else {
+		g.Expect(subscription.ObjectMeta.Finalizers).Should(Equal(expectedFinalizers))
+	}
+}
+
+func ensureSubscriptionMatchesConditionsAndStatus(g *GomegaWithT, subscription eventingv1alpha1.Subscription, expectedConditions []eventingv1alpha1.Condition, expectedStatus bool) {
```

Comment:
Alternative to this (which is more the gomega way I guess) would be something like this (example from the `testing/matchers.go` file).

```go
func HaveAPIRuleSpecRules(ruleMethods []string, accessStrategy, path string) gomegatypes.GomegaMatcher {
	return WithTransform(func(a apigatewayv1alpha1.APIRule) []apigatewayv1alpha1.Rule {
		return a.Spec.Rules
	}, ContainElement(
		MatchFields(IgnoreExtras|IgnoreMissing, Fields{
			"Methods":          ConsistOf(ruleMethods),
			"AccessStrategies": ConsistOf(haveAPIRuleAccessStrategies(accessStrategy)),
			"Gateway":          Equal(constants.ClusterLocalAPIGateway),
			"Path":             Equal(path),
		}),
	))
}
```

It would read like this then (assuming the helper function is called `HaveConditions` instead of `HaveAPIRuleSpecRules`):

```go
			g.Expect(sub.Status.Conditions).Should(HaveConditions(testCase.expectedConditions))
```

Oh I just discovered there is already `HaveCondition` in the same file already :)

```go
func HaveCondition(condition eventingv1alpha1.Condition) gomegatypes.GomegaMatcher {
	return WithTransform(func(s *eventingv1alpha1.Subscription) []eventingv1alpha1.Condition { return s.Status.Conditions }, ContainElement(MatchFields(IgnoreExtras|IgnoreMissing, Fields{
		"Type":    Equal(condition.Type),
		"Reason":  Equal(condition.Reason),
		"Message": Equal(condition.Message),
		"Status":  Equal(condition.Status),
	})))
}
```